framework/src/media/utils/MediaUtils.cpp

```diff
@@ -527,6 +537,41 @@ static bool ts_parsing(unsigned char *buffer, unsigned int bufferSize, audio_typ
 	return buffer_header_parsing(audioES, (unsigned int)audioESLen, *audioType, channel, sampleRate, pcmFormat);
 }
 
+static bool isMpeg2Ts(const unsigned char *buffer, size_t size)
+{
+	if (!buffer) {
+		meddbg("Invalid param, buffer is null!\n");
+		return false;
+	}
+
+	int ret = findTsPacket(buffer, size, true);
+	if (ret < UTILS_ERROR_NONE) {
+		meddbg("Not found TS packets, err %d\n", ret);
+		return false;
+	}
+
+	return true;
+}
+
+static bool isMp3(const unsigned char *buffer, size_t size)
+{
+	// Check ID3v2 tag
+	if ((MP3_ID3V2_HEADER_LENGTH <= size) && (memcmp("ID3", buffer, 3) == 0)) {
+		size_t id3Size = MP3_ID3V2_TAG_SIZE(buffer);
+		if (size <= MP3_ID3V2_HEADER_LENGTH + id3Size) {
+			// not enough data to reach data frame
+			return true;
+		}
+		// Skip ID3v2 tag
+		buffer += MP3_ID3V2_HEADER_LENGTH + id3Size;
+		size -= MP3_ID3V2_HEADER_LENGTH + id3Size;
+	}
+	// try to parse MP3 header
+	unsigned int channel;
+	unsigned int sampleRate;
+	return buffer_header_parsing(buffer, size, AUDIO_TYPE_MP3, &channel, &sampleRate, NULL);
```

Comment:
In this case you can recognize it as a header even if the start of the buffer is not the header.

Reply:
yes, in case there's enough data or there's no ID3v2 tag, buffer_header_parsing will try to sync data frame header.